### PR TITLE
Beef up `Terms#intersect` checks in `CheckIndex`.

### PR DESCRIPTION
**Pull request by jpountz**
Now also testing what happens with a non-null `startTerm`. This found bugs in `DirectPostingsFormat`.

### CODE REVIEW

**Comment by mikemccand** [comment]
> This found bugs in `DirectPostingsFormat`

Whoa, awesome!

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/DirectPostingsFormat.java [hunk 1]
```diff
@@ -1119,12 +1122,14 @@ public DirectIntersectTermsEnum(CompiledAutomaton compiled, BytesRef startTerm)
             }
           }
 
-          final int termOffset = termOffsets[termOrd];
-          final int termLen = termOffsets[1 + termOrd] - termOffset;
+          if (termOrd >= 0) {
```
Comment (mikemccand) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/DirectPostingsFormat.java:1125]: Nice catch!!

**Review by mikemccand — APPROVED** [review]
Love it!  I wonder what flies this new spider web / fangs will catch...
